sql/schema_changer_test.go

```diff
@@ -667,4 +668,42 @@ CREATE UNIQUE INDEX vidx ON t.test (v);
 	if eCount != count {
 		t.Fatalf("read the wrong number of rows: e = %d, v = %d", eCount, count)
 	}
+
+	// Verify that a table delete in the middle of a backfill works properly.
+	// The backfill will terminate in the middle, and the delete will
+	// successfully delete all the table data.
+	//
+	// This test could be made its own test but is placed here to speed up the
+	// testing.
+
+	backfillNotification = make(chan bool)
+	// Run the schema change in a separate goroutine.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		// Start schema change that eventually runs a backfill.
+		_, err := sqlDB.Exec("CREATE UNIQUE INDEX bar ON t.test (v)")
+		if !testutils.IsError(err, "descriptor ID .* not found") {
+			t.Errorf("incorrect error %s", err)
+		}
+		wg.Done()
+	}()
+
+	// Wait until the schema change backfill starts.
+	<-backfillNotification
+
+	// Wait for a short bit to ensure that the backfill has likely progressed
+	// and written some data, but not long enough that the backfill has
+	// completed.
+	time.Sleep(10 * time.Millisecond)
```

Comment:
if you want to get cheeky you can replace this sleep with a KV command filter which watches for writes to the index. And this way you can also stall the backfill at an arbitrary point.